### DISCUSSION TITLE
宝塔安装过程大概记录

### DISCUSSION THREAD

**Discussion by xiamuguizhi** [ArtalkJS/Artalk#46] [Show and tell]
## 前言

我也是刚刚在自己香港服务器搭建成功的，分享一下我用宝塔安装的过程吧！ 

## Docker安装

我们按照官网教程 "https://artalk.js.org/guide/backend/docker.html" 一步步往下走就行！

- 我这里是在宝塔`站点目录`打开终端方便我后期管理编辑

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031115742.png)

打开终端后先执行`镜像拉取`

```bash
docker pull artalk/artalk-go
```

在执行 下面 我圈起来的步骤

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031115941.png) 

这时候我们在看`站点目录` 会多一个`ArtalkGo` 文件夹 在点击进去 有个`conf.yml` 我们 双击 编辑它

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031120209.png)

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031120429.png)

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031120742.png)

我们就先简单配置这两个基本就能运行了  至于其他邮件通知 等等  等搭建好了 在慢慢设置

然后我们继续 在之前的终端 进行下面的步骤

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031121003.png)

在上面步骤 都完成后  执行 `$ docker restart artalk-go` 因为前面我们修改了`conf.yml` 需要重启生效

## 持久化运作

在上面执行完毕后 在终端执行 `docker update --restart=always artalk-go`  好像就是不让这个服务关闭？ 小白不懂 执行就对了。

## 宝塔反向代理

在按照官网教程 `https://artalk.js.org/guide/backend/reverse-proxy.html#%E5%AE%9D%E5%A1%94%E9%9D%A2%E6%9D%BF` 往下走

新建一个站点 就绑定 域名 其他都不需要

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031121419.png)

在如下图

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031121533.png)

若需指定仅 `/api/` 路径转发到后端程序端口，点击 “配置文件” 按钮，添加如图位置红色标记的 `/api/`

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031121727.png)

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031121752.png)

到这里就搞定了！

## 前端配置

```html
<!-- CSS -->
<link href="https://cdn.jsdelivr.net/npm/artalk@2.0.4/dist/Artalk.css" rel="stylesheet">

<!-- JS -->
<script src="https://cdn.jsdelivr.net/npm/artalk@2.0.4/dist/Artalk.js"></script>

<!-- Artalk -->
<div id="Comments"></div>
<script>
  new Artalk({
    el:        '#Comments',
    pageKey:   '<页面链接>',
    pageTitle: '<页面标题>',
    server:    '<后端地址>',
    site:      '<站点名称>',
  })
</script>
```

我是在自己测试服务器搭建的，就简单引用了一下给你们 做个参考 哈~ ` site：`和 `conf.yml` 网站标题相呼应~

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031122101.png)

## Typecho 应该如下就能用了  我没装 还没测试   全部复制以下的代码替换掉 `comments.php` 原先的代码 ok ！就能使

```php
<!-- CSS -->
<link href="https://cdn.jsdelivr.net/npm/artalk@2.0.4/dist/Artalk.css" rel="stylesheet">

<!-- JS -->
<script src="https://cdn.jsdelivr.net/npm/artalk@2.0.4/dist/Artalk.js"></script>

<!-- Artalk -->
<div id="Comments"></div>
<script>
  new Artalk({
    el:        '#Comments',
    pageKey:   '<?php $this->permalink() ?>',
    pageTitle: '<?php $this->title() ?>',
    server:    '你绑定反向代理域名/api/',
    site:      '<?php $this->options->title() ?>',
  })
</script>
```

## 结尾

写的很乱希望你们能看懂，见谅我也是第一次搭建这个... 

测试地址：<del>https://pl.9i3.cn/</del>  准备搭建typecho 测试

![](https://tp.xiamuyourenzhang.cn/qqmd/QQ截图20211031122436.png)

**Comment by xiamuguizhi** [comment]
https://xn--qpru0x.cn/  演示站点

![](https://vkceyugu.cdn.bspapp.com/VKCEYUGU-2fa930c8-feec-4942-ac88-ba3781377bb0/37e5bc30-5e44-4467-98de-5c30f1839d94.png)

**Comment by wulintang** [comment]
写在这里，图片一张都不显示了，江苏特殊用户
